client_option.go

```diff
@@ -1,29 +1,36 @@
 package gax
 
 type ClientOption interface {
-	resolve(*clientSettings)
+	Resolve(*ClientSettings)
 }
 
 type clientOptions []ClientOption
 
-func (opts clientOptions) resolve(s *clientSettings) *clientSettings {
+func (opts clientOptions) Resolve(s *ClientSettings) *ClientSettings {
 	for _, opt := range opts {
-		opt.resolve(s)
+		opt.Resolve(s)
 	}
 	return s
 }
 
-type clientSettings struct {
-	apiName    string
-	apiVersion string
-	endpoint   string
-	scopes     []string
+type ClientSettings struct {
+	APIName    string
+	APIVersion string
+	Endpoint   string
+	Scopes     []string
+}
+
+func (w ClientSettings) Resolve(s *ClientSettings) {
+	s.APIName = w.APIName
+	s.APIVersion = w.APIVersion
+	s.Endpoint = w.Endpoint
+	s.Scopes = append([]string{}, w.Scopes...)
```

Comment:
No, I was wrong. []string{} is fine.

On Mon, May 9, 2016 at 6:55 PM Sai Cheemalapati <notifications@github.com>
wrote:

> In client_option.go
> <https://github.com/googleapis/gax-golang/pull/9#discussion_r62587145>:
>
> > -	apiName    string
> > -	apiVersion string
> > -	endpoint   string
> > -	scopes     []string
> > +type ClientSettings struct {
> > +	APIName    string
> > +	APIVersion string
> > +	Endpoint   string
> > +	Scopes     []string
> > +}
> > +
> > +func (w ClientSettings) Resolve(s *ClientSettings) {
> > +	s.APIName = w.APIName
> > +	s.APIVersion = w.APIVersion
> > +	s.Endpoint = w.Endpoint
> > +	s.Scopes = append([]string{}, w.Scopes...)
>
> When I try append(nil, w.Scopes...), I get:
> first argument to append must be typed slice; have untyped nil
>
> Did I misunderstand?
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/googleapis/gax-golang/pull/9/files/a39373e3a068a88b6b2122f4d7e236c2c86348bd#r62587145>
>